packages/api/portfolios/createPortfolio.test.ts

```diff
@@ -0,0 +1,38 @@
+import { handler } from "./createPortfolio";
+import { ApiGatewayEventParsed } from "../utils/eventHandlingTool";
+import { CloudServiceProvider, DodComponent, IPortfolioCreate } from "../../orm/entity/Portfolio";
+import { ErrorStatusCode, SuccessStatusCode } from "../utils/response";
+import { Context } from "aws-lambda";
+import { ProvisioningStatus } from "../../orm/entity/ProvisionableEntity";
+
+describe("createPortfolio", () => {
+  it.skip("successful operation testing locally only", async () => {
+    const validRequest: ApiGatewayEventParsed<IPortfolioCreate> = {
+      body: {
+        name: "Mock Portfolio",
+        owner: "jane.manager@example.mil",
+        csp: CloudServiceProvider.CSP_A,
+        dodComponents: [DodComponent.ARMY, DodComponent.NAVY],
+        portfolioManagers: ["joe.manager@example.mil", "jane.manager@example.mil"],
+        description: "Mock portfolio description",
+        provisioningStatus: ProvisioningStatus.PENDING,
+      },
+      requestContext: { identity: { sourceIp: "7.7.7.7" } },
+    } as any;
+    const result = await handler(validRequest, {} as Context, () => null);
+    expect(result?.statusCode).toBe(SuccessStatusCode.CREATED);
+  });
+  it("requests with audit properties should cause 400 BAD REQUEST", async () => {
+    const validRequest: ApiGatewayEventParsed<IPortfolioCreate> = {
+      body: {
+        id: "",
+        createdAt: "",
+        updatedAt: "",
+        archivedAt: "",
```

Comment:
Should dummy values be in these?

Reply:
I think their mere presence should trigger the 400.  Being an unexpected additional property shouldn't depend on the value of the property.

The test could be improved to iterate over the collection of disallowed properties.  This will trip on the first one, and not evaluate them all.

Reply:
Yeah, that's true. 

You should be okay without the iteration since the validator checks and gets all the `additionalProperties` that should not be in the request (I did the `it.each()` then remembered the validator capture all invalid properties). It would look something like this.
```typescript
{
  "code": "INVALID_INPUT",
  "message": "Request failed validation",
  "error_map": [
    {
      "instancePath": "/body",
      "schemaPath": "#/properties/body/additionalProperties",
      "keyword": "additionalProperties",
      "params": {
        "additionalProperty": "id"
      },
      "message": "must NOT have additional properties"
    },
    {
      "instancePath": "/body",
      "schemaPath": "#/properties/body/additionalProperties",
      "keyword": "additionalProperties",
      "params": {
        "additionalProperty": "createdAt"
      },
      "message": "must NOT have additional properties"
    },
    {
      "instancePath": "/body",
      "schemaPath": "#/properties/body/additionalProperties",
      "keyword": "additionalProperties",
      "params": {
        "additionalProperty": "updatedAt"
      },
      "message": "must NOT have additional properties"
    },
    {
      "instancePath": "/body",
      "schemaPath": "#/properties/body/additionalProperties",
      "keyword": "additionalProperties",
      "params": {
        "additionalProperty": "archivedAt"
      },
      "message": "must NOT have additional properties"
    }
  ]
}
```